src/main/java/com/inventor/email/EmailControl.java

```diff
@@ -0,0 +1,49 @@
+package com.inventor.email;
+
+import com.inventor.email.settings.EmailSettings;
+import com.inventor.email.service.ServicesE;
+import lombok.AllArgsConstructor;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Date;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+@AllArgsConstructor
+public class EmailControl {
+    EmailSettings working;
+    ServicesE servicesE;
+
+    @GetMapping("/emails")
+    public List<Email> getEmails() {
+        return working.getAll();
+    }
+
+    @PostMapping(consumes = "application/json")
+    public void newEmail(@RequestBody List<Email> email){
+        working.addEmail(email);
+        servicesE.sendAll();
+    }
+
+    @DeleteMapping(value = "/delete/{id}", produces = "application/json")
+    public ResponseEntity<String> deleteEmail(@PathVariable int id) {
+        if (id >= working.getAll().size()) {
```

Comment:
This logic should be moved to the service layer

Reply:
Generally, the more preferrable way to do this will be to throw an Exception in a service and then catch it with another class called controller advice. For this application such a solution will suffice. Here are more structured ways to deal with exceptions https://www.baeldung.com/exception-handling-for-rest-with-spring